util/Report.cc

```diff
@@ -221,9 +237,13 @@ Report::vfileWarn(int /* id */,
 ////////////////////////////////////////////////////////////////
 
 void
-Report::error(int /* id */,
+Report::error(int id,
               const char *fmt, ...)
 {
+  // Skip suppressed messages.
+  if (suppressed_msg_ids_.find(id) != suppressed_msg_ids_.end())
+    return;
```

Comment:
This isn't so simple for errors.
The reason errors throw an exception is because the command will likely seg fault without the check. That is the distinction between warning and error.

Reply:
Ok. My proposed new solution is to add a private `suppressed` variable to `ExceptionMsg` and check that when the exception is caught. Thoughts?